### PR TITLE
Gitlab CI bank

### PR DESCRIPTION
**Pull request by white238**
~~Move all CI to `asmith` and~~ move to the dedicated CI partitions on CZ.

### CODE REVIEW

**Comment by chapman39** [comment]
Does `asmith` need CI permission?

**Comment by white238** [comment]
This is ready to go.

**Review by samuelpmishLLNL — APPROVED** [review]

**Review by jamiebramwell — APPROVED** [review]

File: .gitlab/build_lassen.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # This is the share configuration of jobs for lassen
 .on_lassen:
   variables:
-    ALLOC_COMMAND: lalloc ${ALLOC_NODES} -W ${ALLOC_TIME} -q pdebug -G ${ALLOC_BANK}
+    ALLOC_COMMAND: lalloc ${ALLOC_NODES} -W ${ALLOC_TIME} -q pci -G {ALLOC_BANK}
```
Comment (chapman39) [inline .gitlab/build_lassen.yml]:
```suggestion
    ALLOC_COMMAND: lalloc ${ALLOC_NODES} -W ${ALLOC_TIME} -q pci -G ${ALLOC_BANK}
```

File: .gitlab/build_quartz.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # This is the shared configuration of jobs for quartz
 .on_quartz:
   variables:
-    ALLOC_OPTIONS: "-p pdebug --exclusive=user"
+    ALLOC_OPTIONS: "--res=ci --exclusive=user -A ${ALLOC_BANK}"
```
Comment (chapman39) [inline .gitlab/build_quartz.yml:5]: Is `--res` a new option?

Comment (white238) [reply .gitlab/build_quartz.yml:5]: Yes/no, the command line option has always been there but the functionality is new. it uses the new CI reservation

**Review by chapman39 — APPROVED** [review]

**Pull request merged**